assets/app/views/project.html

```diff
@@ -1,242 +1,239 @@
+
+      <!-- TODO handle multiple services mapping to the same deploymentConfig/deployment/pod -->
+      <section ng-repeat="(serviceId, service) in services" class="components components-group" ng-attr-id="service-{{serviceId}}">
+
+        <div class="osc-object components-panel service" ng-init="numPorts = service.spec.ports.length" kind="Service" resource="service">
+
+          <div class="component-block">
+            <div class="component">
+              <div ng-attr-title="{{service | serviceImplicitDNSName}}" class="component-label">
                 <!-- "service" class is present for e2e tests to check   -->
-                <h2 ng-if="!displayRouteByService[serviceId]">
-                  <a class="service" href="project/{{service.metadata.namespace}}/browse/services/{{serviceId}}">{{serviceId}}</a>
-                </h2>
+                Service <span ng-if="displayRouteByService[serviceId]">: <a class="subtle-link service" href="{{service | navigateResourceURL}}">{{serviceId}}</a></span>
               </div>
 
-              <div class="component meta-data">
-                <div ng-if="numPorts">
-                  <!--
-                  Show only the first two ports if there are many since we don't have much space here.
-                  The full list is visible elsewhere.
-                  -->
-                  <span ng-repeat="portMapping in service.spec.ports | orderBy:'port' | limitTo:2" class="nowrap">
-                    <!-- Group port mappings and allow multiple mappings to stack if needed -->
-                    <span class="port-mappings">
-                      {{portMapping.port}}/{{portMapping.protocol}}&#8201;&#8594;&#8201;{{portMapping.targetPort}}<span ng-if="$index < (numPorts - 1)">, </span></span>
-                  </span>
-                  <span ng-if="numPorts > 2" ng-init="numRemaining = numPorts - 2">
-                    and {{numRemaining}} {{numRemaining == 1 ? "other" : "others"}}
-                  </span>
-                </div>
+              <!-- Show the route if present -->
+              <h2 ng-if="displayRouteByService[serviceId]" ng-init="otherRoutes = (routesByService[serviceId] | hashSize) - 1">
+                <span ng-if="(displayRouteByService[serviceId] | isWebRoute)">
+                  <!-- "route" class is present for e2e tests to check -->
+                  <a href="{{displayRouteByService[serviceId] | routeWebURL}}" class="route" target="_blank">{{displayRouteByService[serviceId] | routeLabel}}</a>
+                </span>
+                <!-- "route" class is present for e2e test -->
+                <span ng-if="!(displayRouteByService[serviceId] | isWebRoute)" class="route">
+                  {{displayRouteByService[serviceId] | routeLabel}}
+                </span>
+                <span class="small" ng-if="otherRoutes">
+                  (and
+                  <a href="project/{{project.metadata.name}}/browse/services#service-{{serviceId}}">
```

Comment:
Ah, I'll hit this.  Judging by the `ng-if="otherRoutes"`, it looks like its actually supposed to be a route (`/project/:project/browse/routes/:route`).